src/WebJobs.Script/Host/DistributedLockManagers/BlobLeaseDistributedLockManager.cs

```diff
@@ -0,0 +1,385 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.Globalization;
+using System.IO;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure;
+using Azure.Storage.Blobs;
+using Azure.Storage.Blobs.Models;
+using Azure.Storage.Blobs.Specialized;
+using Microsoft.Azure.WebJobs.Host;
+using Microsoft.Azure.WebJobs.Logging;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Script
+{
+    /// <summary>
+    /// Provides a CloudBlob lease-based implementation of the <see cref="IDistributedLockManager"/> service for singleton locking.
+    /// This class can be overridden to control where the container comes from.
+    /// The default derived implementation is <see cref="CloudBlobContainerDistributedLockManager"/> which is container based.
+    /// Hosts can provide a derived implementation to leverage the accountName and allow different hosts to share.
+    /// </summary>
```

Comment:
A comment highlighting that this code has been moved from Host.Storage in the SDK repo would be useful (anyone trying to find history for this file will hit a dead-end if they don't realize this)

Reply:
Good call. Will put this comment here.